src/agoradatatools/etl/transform/gene_info.py

```diff
@@ -251,4 +289,7 @@ def transform_gene_info(
     # Make sure there are no N/A Ensembl IDs
     gene_info = gene_info.dropna(subset=["ensembl_gene_id"])
 
+    # Add uniprot ID
+    gene_info = add_uniprot_id_to_gene_info(gene_info, uniprot)
```

Comment:
I would move this to before the .dropna() statement.

Reply:
Actually, now that I look at the whole function -- I would prefer that the uniprot data set gets added in the same way as all the other data sets, for consistency. So, all of the other input data sets have whatever necessary processing done to them pre-line-173. Then line 173 begins a loop over all data sets to do outer joins on `ensembl_gene_id`. So your `add_uniprot_id_to_gene_info()` function should actually be a function that creates `collapsed_uniprot` and returns it, and then that df gets added to the loop at line 173. Does that make sense?
This also simplifies testing that function vs testing the whole transform, in my opinion.

Reply:
This makes sense! Thank you for catching it! In an effort to cause the lest amount of code changes (and therefore testing) I missed this opportunity for better integration.

Reply:
Also just FYI, everything is an outer join instead of a left join because while we try to keep `gene_metadata` up to date to have all genes in all data sets, it is possible that adding a new data set or re-running pre-processing on one of the datasets we process could introduce genes that aren't in `gene_metadata` yet, and we wanted those genes to get added.

Although, now I'm wondering if we _should_ do left joins instead and throw a warning or error to update `gene_info`. Not your problem to fix right now but, worth discussing some time.